ibc/utils.go

```diff
@@ -124,3 +128,45 @@ func GetSentCoin(rawDenom, rawAmt string) sdk.Coin {
 		Amount: amount,
 	}
 }
+
+// GetDenomTrace returns the denomination trace from the corresponding IBC denomination. If the
+// denomination is not an IBC voucher or the trace is not found, it returns an error.
+func GetDenomTrace(
+	transferKeeper transferkeeper.Keeper,
+	ctx sdk.Context,
+	denom string,
+) (transfertypes.DenomTrace, error) {
+	if !strings.HasPrefix(denom, "ibc/") {
+		return transfertypes.DenomTrace{}, errorsmod.Wrapf(ErrNoIBCVoucherDenom, denom)
+	}
+
+	hash, err := transfertypes.ParseHexHash(denom[4:])
+	if err != nil {
+		return transfertypes.DenomTrace{}, err
+	}
+
+	denomTrace, found := transferKeeper.GetDenomTrace(ctx, hash)
+	if !found {
+		return transfertypes.DenomTrace{}, ErrDenomTraceNotFound
+	}
+
+	return denomTrace, nil
+}
+
+// GetDenomDecimals returns the number of decimals of an IBC coin
+// depending on the prefix of the base denomination
+func GetDenomDecimals(baseDenom string) (uint8, error) {
```

Comment:
it might be clearer if we rename this to `DeriveDecimalsFromDenom` - the name as it is sounds more like we are actually getting the decimals from the coin metadata